token/services/network/processor/db/tokenstore.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package db
+
+import (
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/events"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/hyperledger-labs/fabric-token-sdk/token"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/network/processor"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/tokendb"
+	token2 "github.com/hyperledger-labs/fabric-token-sdk/token/token"
+	"github.com/pkg/errors"
+)
+
+var logger = flogging.MustGetLogger("token-sdk.network.processor")
+
+type TokenStore struct {
+	notifier events.Publisher
+	tokenDB  *tokendb.DB
+	tmsID    token.TMSID
+}
+
+func NewTokenStore(notifier events.Publisher, tokenDB *tokendb.DB, tmsID token.TMSID) (*TokenStore, error) {
+	return &TokenStore{
+		notifier: notifier,
+		tokenDB:  tokenDB,
+		tmsID:    tmsID,
+	}, nil
+}
+
+func (t *TokenStore) DeleteToken(ns string, txID string, index uint64, rws processor.RWSet, deletedBy string) error {
+	tok, owners, err := t.tokenDB.OwnersOf(ns, txID, index)
+	if err != nil {
+		return errors.WithMessagef(err, "failed to get owners for token [%s:%s:%d]", ns, txID, index)
+	}
+	err = t.tokenDB.Delete(ns, txID, index, deletedBy)
+	if err != nil {
+		if tok == nil {
+			logger.Debugf("nothing further to delete for [%s:%d]", txID, index)
+			return nil
+		}
+		return errors.WithMessagef(err, "failed to delete token [%s:%s:%d]", ns, txID, index)
+	}
+	for _, id := range owners {
+		logger.Debugf("post new delete-token event")
+		t.Notify(processor.DeleteToken, t.tmsID, id, tok.Type, txID, index)
+	}
+	return nil
+}
+
+func (t *TokenStore) StoreToken(ns string, txID string, index uint64, tok *token2.Token, rws processor.RWSet, tokenOnLedger []byte, tokenOnLedgerMetadata []byte, ids []string) error {
+	err := t.tokenDB.StoreOwnerToken(
+		tokendb.TokenRecord{
+			TxID:           txID,
+			Index:          index,
+			Namespace:      ns,
+			IssuerRaw:      nil,
+			OwnerRaw:       tok.Owner.Raw,
+			Ledger:         tokenOnLedger,
+			LedgerMetadata: tokenOnLedgerMetadata,
+			Quantity:       tok.Quantity,
+			Type:           tok.Type,
+			Amount:         0,
```

Comment:
Should we still add the Amount as number to the function signature? We have it for issuedTokens and it'd allow for some nice queries if we add it (get all balances, for instance, and business reporting). By the way shall we also add a StoredAt? That kind of info is always good for debugging or auditability.

Reply:
We could introduce a limit to the size of a transaction... 64 bit is already 18 billion billion, do we need more?